exporter/prometheusexporter/prometheus.go

```diff
@@ -37,19 +37,20 @@ type prometheusExporter struct {
 	shutdownFunc func() error
 }
 
-var _ consumer.MetricsConsumerOld = (*prometheusExporter)(nil)
-
 func (pe *prometheusExporter) Start(_ context.Context, _ component.Host) error {
 	return nil
 }
 
-func (pe *prometheusExporter) ConsumeMetricsData(ctx context.Context, md consumerdata.MetricsData) error {
-	merged := make(map[string]*metricspb.Metric)
-	for _, metric := range md.Metrics {
-		merge(merged, metric)
-	}
-	for _, metric := range merged {
-		_ = pe.exporter.ExportMetric(ctx, md.Node, md.Resource, metric)
+func (pe *prometheusExporter) ConsumeMetrics(ctx context.Context, md pdata.Metrics) error {
+	ocmds := pdatautil.MetricsToMetricsData(md)
```

Comment:
Out of curiousity: what's the long term plan for this? Having to convert the data as the very first step in the "new" metric exporter is pretty confusing.

Reply:
The long term is to change to use the new data directly without oc as intermediate step. Right now focus on changing everything to use the new interfaces so at least we are not everywhere in this transition state and service and config code can be changed to only work with the new interfaces